### PR TITLE
dbus: update bindings with newly introspected ones and other maintainance

### PR DESCRIPTION
**Pull request by hnez**
I was curious if anything relevant changed in the API bindings.
This was not the case, but now that we have the new bindings we might as well use them 🤷‍♂️.

Also do some other cleanups while we are at it.

Related Pull Requests
---------------

This PR is based on another PR that should be merged first:

- [x] tacd #44

### CODE REVIEW

File: src/dbus/hostname.rs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
```
Comment (KarlK90) [inline src/dbus/hostname.rs:16]:
```suggestion
// This file is part of tacd, the LXA TAC system daemon
// Copyright (C) 2023 Pengutronix e.K.
// SPDX-License-Identifier: GPL-2.0-only
```

How about machine readable and concise license headers?

**Review by KarlK90 — APPROVED** [review]
LGTM, maybe consider moving to SPDX license headers for machine readability?

File: src/dbus/hostname.rs (hunk 1 above)
Comment (hnez) [reply src/dbus/hostname.rs:16]: Maybe. But If we do that we should change it in all of the existing files as well.

Comment (KarlK90) [reply src/dbus/hostname.rs:16]: Sure, I'm happy to review that one ;-)

**Pull request merged**